### ISSUE TITLE
How to get the editor instance ?

### ISSUE THREAD

**Issue by abhishek97** [mike-north/ember-monaco#51]
I see that current the only way to get the editor instance is to get `editor` property on iframe's contentWindow. Is there a better way ?, like a onReady action to passed on as an argument to the component?

**Comment by mike-north** [comment]
It's been a while since I wrote this, so I'll have to dig in to answer this more thoroughly 

> get the editor instance is to get editor property on iframe's contentWindow

This is how I'd do it. I like your idea of an `onReady` action, but it might encourage leaking this component's concerns outward. I'd favor building actions/properties that allow the component to be the only thing that knows about the editor

**Comment by abhishek97** [comment]
However, there are usercases where we need the editor, like in my case, I need to connect monaco with [firepad](https://firepad.io/). Currently I have this

```js
didInsertElement () {
    this._super(...arguments)
    const monacoIframe = document.querySelector('iframe[src*="ember-monaco"]')
    monacoIframe.addEventListener('load', () => {
      const iframeWindow = monacoIframe.contentWindow
      
      // Get the editor reference and set monaco global
      this.editor = iframeWindow.editor
      window.monaco = iframeWindow.monaco
      
      // setup firepad service
      const firepad = this.firepad
      firepad.set("editor", this.editor)
    })
```

Not the most elegant solution, but works.  Have a look at https://github.com/dfreeman/ember-ace for ace editor. A `onReady` action is required for anyone who want to use external lib with editor.

**Comment by mike-north** [comment]
aha, yes I'd agree that you just want the editor in your case. I'd welcome a PR to expose this from the component, so that subclasses may consume it more easily

**Comment by mike-north** [comment]
Fixed by #53
